src/main/java/nextstep/ladder/domain/Ladder.java

```diff
@@ -0,0 +1,57 @@
+package nextstep.ladder.domain;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class Ladder {
+
+    private List<Line> lines;
+
+    private Ladder(List<Line> lines) {
+        this.lines = lines;
+    }
+
+    public static Ladder of(int height, int countOfPerson, Line... lines) {
+        return of(height, countOfPerson, Arrays.asList(lines));
+    }
+
+    public static Ladder of(int height, int countOfPerson, List<Line> lines) {
+        validateHeight(height, lines);
+        validateSize(countOfPerson, lines);
+        return new Ladder(lines);
+    }
+
+    private static void validateHeight(int height, List<Line> lines) {
+        if (lines.size() != height) {
+            throw new IllegalArgumentException("사다리 높이와 일치하지 않습니다.");
+        }
+    }
+
+    private static void validateSize(int countOfPerosn, List<Line> lines) {
+        Optional<Line> maybeLine = lines.stream()
+                .filter(line -> !line.isSameSize(countOfPerosn - 1))
+                .findAny();
+        if (maybeLine.isPresent()) {
+            throw new IllegalArgumentException("Point 개수가 올바르지 않습니다.");
+        }
+    }
+
+    public List<Line> getLines() {
+        return lines;
+    }
```

Comment:
꼭 필요할지 한번만 더 생각해주세요!
그리고 컬렉션 객체를 리턴할 땐, 아래와 같이 가급적 불변 컬렉션으로 감싸서 리턴해주세요.

```suggestion
    public LIst<Line> getLines() {
        return Collections.unmodifiableList(lines);
}

Reply:
해당 getter 메소드는 뷰 레이어에서만 사용하려고 했습니다. getter 대신에 다른 메소드를 사용할 수 있지만 결국엔 String을 반환하는 메소드가 되어버려서 뷰에 의존적인 코드가 될 것 같아요. 🤔

피드백 받고 일종의 DTO 역할을 수행하는 enum(`Pipe`) 정의해서 DTO를 반환하도록 변경해봤습니다! 그런데 결국엔 DTO의 getter를 호출하게 되더라구요... 좋은 방법이 있을까요?

Reply:
우선 리뷰가 늦어져 정말 죄송합니다 ㅠㅠㅠㅠㅠ
코드 코멘트 통해 답변 드릴게요!